platform/darwin/test/MGLLightTest.mm

```diff
@@ -0,0 +1,211 @@
+        XCTAssertEqual(mglLight.positionTransiton.duration, defaultTransition.duration);
+        
+        auto lightFromMGLlight = [mglLight mbglLight];
+        
+        XCTAssertEqual(positionArray, lightFromMGLlight.getPosition().asConstant().getSpherical());
+        auto positionTransition = lightFromMGLlight.getPositionTransition();
+        XCTAssert(positionTransition.delay && MGLTimeIntervalFromDuration(*positionTransition.delay) == defaultTransition.delay);
+        XCTAssert(positionTransition.duration && MGLTimeIntervalFromDuration(*positionTransition.duration) == defaultTransition.duration);
+        
+        defaultPosition = MGLLightPositionMake(6, 180, 90);
+        MGLStyleValue<NSValue *> *positionStyleValue = [MGLStyleValue<NSValue *> valueWithRawValue:[NSValue valueWithMGLLightPosition:defaultPosition]];
+        mglLight.position = positionStyleValue;
+        mglLight.positionTransiton = transition;
+        
+        NSAssert([mglLight.position isKindOfClass:[MGLConstantStyleValue class]], @"mglLight.position isn’t a MGLConstantStyleValue.");
+        positionValue = ((MGLConstantStyleValue *)mglLight.position).rawValue;
+        
+        XCTAssert(defaultPosition.radial == positionValue.MGLLightPositionValue.radial);
+        XCTAssert(defaultPosition.azimuthal == positionValue.MGLLightPositionValue.azimuthal);
+        XCTAssert(defaultPosition.polar == positionValue.MGLLightPositionValue.polar);
+        XCTAssertEqual(mglLight.positionTransiton.delay, transition.delay);
+        XCTAssertEqual(mglLight.positionTransiton.duration, transition.duration);
+        
+        lightFromMGLlight = [mglLight mbglLight];
+        
+        positionArray = { { 6, 180, 90 } };
+        mbgl::style::Position position = { positionArray };
+        mbgl::style::PropertyValue<mbgl::style::Position> positionProperty = { position };
+        light->setPosition(positionProperty);
+        light->setPositionTransition(transitionOptions);
+
+        XCTAssertEqual(positionArray, lightFromMGLlight.getPosition().asConstant().getSpherical());
+        positionTransition = lightFromMGLlight.getPositionTransition();
+        XCTAssert(positionTransition.delay && MGLTimeIntervalFromDuration(*positionTransition.delay) == transition.delay);
+        XCTAssert(positionTransition.duration && MGLTimeIntervalFromDuration(*positionTransition.duration) == transition.duration);
+
+    }
+    
+    // color
+    {
+        auto light = new mbgl::style::Light();
+        MGLLight *mglLight = [[MGLLight alloc] initWithMBGLLight:light];
+        NSAssert([mglLight.color isKindOfClass:[MGLConstantStyleValue class]], @"mglLight.color isn’t a MGLConstantStyleValue.");
+        MGLColor *colorValue = ((MGLConstantStyleValue *)mglLight.color).rawValue;
+        auto color = light->getDefaultColor();
+        const CGFloat *colorComponents = CGColorGetComponents(colorValue.CGColor);
```

Comment:
That's because the color space for whiteColor is a grayscale color space rather than the RGB color space. We have methods for converting between NSColor/UIColor and mbgl::Color.